src/mlpack/core/tree/cosine_tree/cosine_tree.cpp

```diff
@@ -17,12 +17,12 @@
 namespace mlpack {
 namespace tree {
 
-CosineTree::CosineTree(const arma::mat& dataset) :
+CosineTree::CosineTree(arma::mat* dataset) :
```

Comment:
Let's not change the API here.  Holding a pointer to the dataset internally is just fine, but take a `const arma::mat&` as the input please. :+1:  (You may need to hold a const pointer internally, then.)